### ISSUE TITLE
Instant crash on MacOS when window is shrunk vertically

### ISSUE THREAD

**Issue by pixelgaming579** [parasyte/pixels#344]
I am consistently getting an instant crash on the latest versions of both Winit (0.28.2) and Pixels (0.11.0), but only when shrinking the window vertically. More precisely, the window allows me to continue resizing it, but has no other functionality will instantly disappear as soon as I stop resizing it. This bug doesn't occur in older versions of Winit (0.27.5) and Pixels (0.10.0) with the exact same code, shown below:

```
use pixels::{SurfaceTexture, Pixels};
use winit::{event_loop::EventLoop, event::{Event, WindowEvent}, window::Window};

fn main() {
    let event_loop = EventLoop::new();
    let window = winit::window::WindowBuilder::new()
    .with_title("Hello Winit/Pixels!")
    .build(&event_loop).unwrap();

    let mut pixel_buf = {
        let size = window.inner_size();
        let texture = SurfaceTexture::new(size.width, size.height, &window);
        Pixels::new(size.width, size.height, texture).unwrap()
    };

    event_loop.run(move |event, _, control_flow| {
        control_flow.set_poll();

        // Render window.
        match event {
            Event::MainEventsCleared | Event::RedrawRequested(_) => {
                render(&window, &mut pixel_buf);
                if let Err(err) = pixel_buf.render() {
                    panic!("Renderer encountered an error! : {}", err);
                }
            },
            Event::WindowEvent { event: WindowEvent::Resized(s), .. } => {
                println!("{:?}", s);
                pixel_buf.resize_surface(s.width, s.height);
                pixel_buf.resize_buffer(s.width, s.height);
                window.request_redraw();
            },
            Event::WindowEvent { event: WindowEvent::CloseRequested, .. } => {
                control_flow.set_exit();
            },
            _ => ()
        }
    });
}

fn render(window: &Window, frame_buf: &mut Pixels) { // Renders a simple black to white gradient from top-left to bottom-right.
    let win_size = window.inner_size();
    let max_dist = ((win_size.width as f64).powi(2) + (win_size.height as f64).powi(2)).sqrt();
    for (i, pixel) in frame_buf.get_frame_mut().chunks_exact_mut(4).enumerate() {
        let x = (i % win_size.width as usize) as f64;
        let y = (i / win_size.width as usize) as f64;
        let dist = (x.powi(2) + y.powi(2)).sqrt();
        let c = (255. * dist / max_dist) as u8;
        pixel.copy_from_slice(&[c,c,c,255]);
    }
}
```

Here's the error and backtrace:
```
thread 'main' panicked at 'assertion failed: min <= max', /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/num/f32.rs:1394:9
stack backtrace:
   0: rust_begin_unwind
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:142:14
   2: core::panicking::panic
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:48:5
   3: core::f32::<impl f32>::clamp
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/num/f32.rs:1394:9
   4: pixels::renderers::ScalingMatrix::new
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/pixels-0.11.0/src/renderers.rs:234:21
   5: pixels::Pixels::resize_surface
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/pixels-0.11.0/src/lib.rs:343:39
   6: ddd_engine::main::{{closure}}
             at ./src/main.rs:29:17
   7: <winit::platform_impl::platform::app_state::EventLoopHandler<T> as winit::platform_impl::platform::app_state::EventHandler>::handle_nonuser_event::{{closure}}
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.28.2/src/platform_impl/macos/app_state.rs:96:17
   8: winit::platform_impl::platform::app_state::EventLoopHandler<T>::with_callback
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.28.2/src/platform_impl/macos/app_state.rs:70:13
   9: <winit::platform_impl::platform::app_state::EventLoopHandler<T> as winit::platform_impl::platform::app_state::EventHandler>::handle_nonuser_event
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.28.2/src/platform_impl/macos/app_state.rs:91:9
  10: winit::platform_impl::platform::app_state::Handler::handle_nonuser_event
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.28.2/src/platform_impl/macos/app_state.rs:199:21
  11: winit::platform_impl::platform::app_state::AppState::cleared
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.28.2/src/platform_impl/macos/app_state.rs:384:13
  12: winit::platform_impl::platform::observer::control_flow_end_handler::{{closure}}
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.28.2/src/platform_impl/macos/observer.rs:79:21
  13: winit::platform_impl::platform::observer::control_flow_handler::{{closure}}
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.28.2/src/platform_impl/macos/observer.rs:41:9
  14: std::panicking::try::do_call
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/std/src/panicking.rs:492:40
  15: ___rust_try
  16: std::panicking::try
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/std/src/panicking.rs:456:19
  17: std::panic::catch_unwind
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/std/src/panic.rs:137:14
  18: winit::platform_impl::platform::event_loop::stop_app_on_panic
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.28.2/src/platform_impl/macos/event_loop.rs:245:11
  19: winit::platform_impl::platform::observer::control_flow_handler
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.28.2/src/platform_impl/macos/observer.rs:39:5
  20: winit::platform_impl::platform::observer::control_flow_end_handler
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.28.2/src/platform_impl/macos/observer.rs:74:9
  21: <unknown>
  22: <unknown>
  23: <unknown>
  24: <unknown>
  25: <unknown>
  26: <unknown>
  27: <unknown>
  28: <unknown>
  29: <unknown>
  30: <unknown>
  31: <unknown>
  32: <unknown>
  33: <unknown>
  34: <unknown>
  35: <unknown>
  36: <unknown>
  37: <unknown>
  38: <unknown>
  39: <(A,) as objc2::message::MessageArguments>::__invoke
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/objc2-0.3.0-beta.3/src/message/mod.rs:390:26
  40: objc2::message::platform::send_super_unverified::{{closure}}
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/objc2-0.3.0-beta.3/src/message/apple/mod.rs:60:33
  41: objc2::message::conditional_try
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/objc2-0.3.0-beta.3/src/message/mod.rs:30:5
  42: objc2::message::platform::send_super_unverified
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/objc2-0.3.0-beta.3/src/message/apple/mod.rs:60:14
  43: objc2::message::MessageReceiver::send_super_message
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/objc2-0.3.0-beta.3/src/message/mod.rs:237:46
  44: objc2::message::MessageReceiver::__send_super_message_static
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/objc2-0.3.0-beta.3/src/message/mod.rs:250:18
  45: winit::platform_impl::platform::app::WinitApplication::send_event
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.28.2/src/platform_impl/macos/app.rs:38:26
  46: <unknown>
  47: <unknown>
  48: <() as objc2::message::MessageArguments>::__invoke
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/objc2-0.3.0-beta.3/src/message/mod.rs:390:26
  49: objc2::message::platform::send_unverified::{{closure}}
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/objc2-0.3.0-beta.3/src/message/apple/mod.rs:36:33
  50: objc2::message::conditional_try
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/objc2-0.3.0-beta.3/src/message/mod.rs:30:5
  51: objc2::message::platform::send_unverified
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/objc2-0.3.0-beta.3/src/message/apple/mod.rs:36:14
  52: objc2::message::MessageReceiver::send_message
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/objc2-0.3.0-beta.3/src/message/mod.rs:195:46
  53: winit::platform_impl::platform::appkit::application::NSApplication::run
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/objc2-0.3.0-beta.3/src/macros/__rewrite_self_arg.rs:42:21
  54: winit::platform_impl::platform::event_loop::EventLoop<T>::run_return::{{closure}}
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.28.2/src/platform_impl/macos/event_loop.rs:220:22
  55: objc2::rc::autorelease::autoreleasepool
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/objc2-0.3.0-beta.3/src/rc/autorelease.rs:313:5
  56: winit::platform_impl::platform::event_loop::EventLoop<T>::run_return
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.28.2/src/platform_impl/macos/event_loop.rs:211:25
  57: winit::platform_impl::platform::event_loop::EventLoop<T>::run
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.28.2/src/platform_impl/macos/event_loop.rs:190:25
  58: winit::event_loop::EventLoop<T>::run
             at /Users/home/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.28.2/src/event_loop.rs:305:9
  59: ddd_engine::main
             at ./src/main.rs:16:5
  60: core::ops::function::FnOnce::call_once
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/ops/function.rs:248:5
```

**Comment by AndroGR** [comment]
I can verify this happened to me too. If I remove the call to
```rust
pixels.resize_buffer(...);
```
Then this goes away. However you lose the ability to resize the buffer.

**Comment by pixelgaming579** [comment]
> I can verify this happened to me too. If I remove the call to
> 
> ```rust
> pixels.resize_buffer(...);
> ```
> 
> Then this goes away. However you lose the ability to resize the buffer.

However it isn't much of a fix (and I don't know if there is a way to get the buffer's size instead of the window's).

![Screenshot after resizing window](https://user-images.githubusercontent.com/66503946/225747075-387e5c67-b775-4331-a76a-0b7e18eac28a.png)

**Comment by parasyte** [comment]
This looks like a duplicate of #330, which was fixed in #332. It isn't released yet, but you can use the current head of the `main` branch in your Cargo.toml like this, and the problem will go away:

```toml
[dependencies]
pixels = { git = "https://github.com/parasyte/pixels.git", rev = "899e27bafa6eac42ced144878f5c946e49139536" }
winit = "0.28"
```

**Comment by pixelgaming579** [comment]
> This looks like a duplicate of #330, which was fixed in #332. It isn't released yet, but you can use the current head of the `main` branch in your Cargo.toml like this, and the problem will go away:
> 
> ```toml
> [dependencies]
> pixels = { git = "https://github.com/parasyte/pixels.git", rev = "899e27bafa6eac42ced144878f5c946e49139536" }
> winit = "0.28"
> ```
> 
> > and I don't know if there is a way to get the buffer's size instead of the window's
> 
> You can get the buffer size with [`pixels.context().texture_extent`](https://docs.rs/pixels/0.11.0/pixels/struct.PixelsContext.html#structfield.texture_extent).

Working perfectly now, thanks.
